### PR TITLE
Make Box.Inflate work like System.Drawing.Rectangle.Inflate

### LINKED ISSUE #1616: Reimplement Box.Inflate with System.Drawing matching functionality
### Description

In #1511 and #1615 `Box.Inflate` got marked as `Obsolete` pending a change to the functionality of the function to match `System.Drawing.Rectangle.Inflate`.

We should add a remark in the documentation of `Box.Inflate` that it used to do something else so that someone who is upgrading from `4.7.7 -> 4.8.2` has any chance to find the issue when debugging.

### PR DESCRIPTION
**Pull request by MV10**
Re-implementation of https://github.com/opentk/opentk/commit/e61afa31dcbceebed54120a570d20fb1f2869f33 which makes the `Box.Inflate` methods more like `System.Drawing.Rectangle.Inflate`

Fixes #1616 

* Altered `Inflate` and `Inflated` for `Box2`, `Box2d`, `Box2i`, `Box3`, `Box3d`, and `Box3i`
* Updated XML API docs to reference `Extend` and `Extended` if previous functionality is needed
* Created `Box2` and `Box3` unit tests for `Inflate`
* Created `Box2` and `Box3` unit tests for `Extend`

Testing relied on the unit tests, however `Inflate` is currently failing due to rounding differences between the F# test and the C# implementation; details in [this](https://github.com/opentk/opentk/issues/1616#issuecomment-1763412383) post. As the code is trivial, I'm sure it actually works, so I guess this is a question of your expectations about test automation.

Edit: Per discussions on Discord, I've commented out those tests (with details).

### CODE REVIEW

File: src/OpenTK.Mathematics/Geometry/Box2.cs [hunk 1]
```diff
@@ -261,28 +261,29 @@ public Box2 Scaled(Vector2 scale, Vector2 anchor)
         }
 
         /// <summary>
-        /// Inflate this Box2 to encapsulate a given point.
+        /// Inflates this Box2 by the given size in all directions. A negative size will shrink the box to a maximum of -HalfSize.
+        /// Use the <see cref="Extend"/> method for the point-encapsulation functionality in earlier API versions.
```
Comment (NogginBops) [inline src/OpenTK.Mathematics/Geometry/Box2.cs:265]: I think "earlier API versions" should be more specific about versions.
Something like `pre 4.8.0` or whichever function deprecated the old functionality.

File: src/OpenTK.Mathematics/Geometry/Box2.cs [hunk 2]
```diff
@@ -261,28 +261,29 @@ public Box2 Scaled(Vector2 scale, Vector2 anchor)
...
-            _max = Vector2.ComponentMax(_max, point);
+            size = Vector2.ComponentMax(size, -HalfSize);
+            _min -= size;
+            _max += size;
         }
 
         /// <summary>
-        /// Inflate this Box2 to encapsulate a given point.
+        /// Inflates this Box2 by the given size in all directions. A negative size will shrink the box to a maximum of -HalfSize.
+        /// Use the <see cref="Extended"/> method for the point-encapsulation functionality in earlier API versions.
```
Comment (NogginBops) [inline src/OpenTK.Mathematics/Geometry/Box2.cs:277]: "earlier API version", see other comment.

File: tests/OpenTK.Tests/Box3Tests.fs [hunk 3]
```diff
@@ -158,25 +158,42 @@ module Box3 =
             
     [<Properties(Arbitrary = [|typeof<OpenTKGen>|])>]
     module Inflate =
+
+        // See #1616 and PR 1662 -- it appears the test-runner generates F# 64-bit float inputs and
+        // C# 32-bit floats sometimes cause equivalency failures. Could possibly be fixed by updating
+        // Assertions.fs with Equal overloads that accept the precision argument, but per discussion on
+        // Discord the math tests may need re-thinking. In that case, how much variance is acceptable?
```
Comment (NogginBops) [inline tests/OpenTK.Tests/Box3Tests.fs:165]: I'll investigate this before merging this PR.

File: src/OpenTK.Mathematics/Geometry/Box2.cs (hunk 1 above)
Comment (MV10) [reply src/OpenTK.Mathematics/Geometry/Box2.cs]: Updated all references to `OpenTK version 4.8.1 and earlier`

File: tests/OpenTK.Tests/Box3Tests.fs (hunk 3 above)
Comment (NogginBops) [reply tests/OpenTK.Tests/Box3Tests.fs]: F# was not doing anything with 64-bit floats. It was floating point imprecisions in the Inflate implementation that made the calculations not equal each other.

**Comment by NogginBops** [comment]
Threw in some fixes to compilation issues in the other test projects while fixing the floating point issues.
CI passes. Merging!

**Pull request merged**